spec/02-integration/05-proxy/10-server_tokens_spec.lua

```diff
@@ -0,0 +1,286 @@
+local helpers = require "spec.helpers"
+local constants = require "kong.constants"
+
+
+local default_server_header = _KONG._NAME.."/".._KONG._VERSION
+
+
+local function start(config)
+  return function()
+    helpers.dao.apis:insert {
+      name = "api-1",
+      upstream_url = "http://localhost:9999/headers-inspect",
+      hosts = "inexistent.com",
+    }
+
+    config = config or {}
+    config.nginx_conf = "spec/fixtures/custom_nginx.template"
+
+    assert(helpers.start_kong(config))
+  end
+end
+
+
+describe("Server Tokens", function()
+  local client
+
+  before_each(function()
+    client = helpers.proxy_client()
+  end)
+
+  after_each(function()
+    if client then
+      client:close()
+    end
+  end)
+
+  describe("(with default configration values)", function()
+
+    setup(start {
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+    })
+
+    teardown(helpers.stop_kong)
+
+    it("should return Kong 'Via' header but not change the 'Server' header", function()
+      local res = assert(client:send {
+        method  = "GET",
+        path    = "/get",
+        headers = {
+          host  = "inexistent.com"
+        }
+      })
+
+      assert.response(res).has.status(200)
+      assert.not_equal(default_server_header, res.headers["server"])
+      assert.equal(default_server_header, res.headers["via"])
+    end)
+
+    it("should return Kong 'Server' header but not the Kong 'Via' header", function()
+      local res = assert(client:send {
+        method  = "GET",
+        path    = "/get",
+        headers = {
+          host  = "404.com"
+        }
+      })
+
+      assert.response(res).has.status(404)
+      assert.response(res).has_not.header "via"
```

Comment:
Those assertions are locking us in and will make it harder for us to switch to another mocking service. We should stop using them, as already raised in the style guide PR.